sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -704,22 +1156,138 @@ pi_result piMemRetain(pi_mem) {
 }
 
 pi_result piMemRelease(pi_mem Mem) {
-  (void)Mem;
-  DIE_NO_IMPLEMENTATION;
+  if (Mem->getMemType() == PI_MEM_TYPE_BUFFER) {
+    _pi_buffer *pi_buf = static_cast<_pi_buffer *>(Mem);
+    int result = Mem->Context->CmDevicePtr->DestroySurface(pi_buf->CmBufferPtr);
+
+    if (result != cm_support::CM_SUCCESS) {
+      return PI_INVALID_MEM_OBJECT;
+    }
+  } else if (Mem->getMemType() == PI_MEM_TYPE_IMAGE2D) {
+    _pi_image *pi_image = static_cast<_pi_image *>(Mem);
+    int result =
+        Mem->Context->CmDevicePtr->DestroySurface(pi_image->CmSurfacePtr);
+
+    if (result != cm_support::CM_SUCCESS) {
+      return PI_INVALID_MEM_OBJECT;
+    }
+  } else if (Mem->getMemType() == PI_MEM_TYPE_IMAGE2D) {
+    _pi_image *pi_img = static_cast<_pi_image *>(Mem);
+    int result =
+        Mem->Context->CmDevicePtr->DestroySurface(pi_img->CmSurfacePtr);
+    if (result != cm_support::CM_SUCCESS) {
+      return PI_INVALID_MEM_OBJECT;
+    }
+  } else {
+    return PI_INVALID_MEM_OBJECT;
+  }
+
   return PI_SUCCESS;
 }
 
+cm_support::CM_SURFACE_FORMAT
+piImageFormatToCmFormat(const pi_image_format *piFormat) {
```

Comment:
was this function added to PI interface really? if not, then its name should not start from `pi`.

Reply:
Will rename it.

Reply:
Done.